### PR TITLE
Added Subscriber tests for Cleanup on Unbind

### PR DESCRIPTION
**Pull request by pcalin**

### CODE REVIEW

File: src/NetMQ/NetMQ.nuspec [hunk 1]
```diff
@@ -16,9 +16,6 @@
     </releaseNotes>
     <copyright>Copyright 2016</copyright>
     <tags>ZeroMQ 0MQ CLRZMQ NetMQ Messaging ZMQ transport distributed</tags>
-    <dependencies>
-      <dependency id="AsyncIO" version="0.1.18.0" />
```
Comment (drewnoakes) [inline src/NetMQ/NetMQ.nuspec]: Why did you remove AsyncIO?

File: src/NetMQ/packages.NetMQ.config [hunk 2]
```diff
@@ -1,5 +1,5 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
   <package id="AsyncIO" version="0.1.18.0" targetFramework="net40" />
-  <package id="JetBrains.Annotations" version="10.0.0" targetFramework="net40" />
+  <package id="JetBrains.Annotations" version="10.0.0" targetFramework="net40" developmentDependency="true"/>
```
Comment (drewnoakes) [inline src/NetMQ/packages.NetMQ.config]: Interesting and probably worthwhile though has no effect on our current way of packing for NuGet as we list dependencies in the nuspec file.

http://nuget.codeplex.com/workitem/1956

For completeness this might also be included in the .NET 3.5 .config file too.

File: src/NetMQ/NetMQ.nuspec (hunk 1 above)
Comment (pcalin) [inline src/NetMQ/NetMQ.nuspec]: I did not. I made sure nuget will properly use the config in the project. AsyncIO becomes a dependency because it is in the config.

File: src/NetMQ/packages.NetMQ.config (hunk 2 above)
Comment (pcalin) [inline src/NetMQ/packages.NetMQ.config]: See above. Because I removed the explicit dependencies from nuspec, nuget will now correctly bring in the real dependencies ignoring all developer dependencies.

I will add the same for .Net 3.5

**Comment by pcalin** [comment]
Is there anything else holding this back, guys?
